File: docs/data-tools/codesnippet/CSharp/save-data-in-a-transaction_4.cs
        private void AddNewCustomers()
        {
            NorthwindDataSet.CustomersDataTable newCustomers;
            newCustomers = (NorthwindDataSet.CustomersDataTable)
                northwindDataSet.Customers.GetChanges(DataRowState.Added);

            if (newCustomers != null)
            {
                try
                {
                    customersTableAdapter.Update(newCustomers);
                }
                catch (System.Exception ex)
                {
                    MessageBox.Show("AddNewCustomers Failed");
                }
            }
        }

File: docs/data-tools/codesnippet/VisualBasic/save-data-in-a-transaction_4.vb
    Private Sub AddNewCustomers()

        Dim newCustomers As NorthwindDataSet.CustomersDataTable
        newCustomers = CType(NorthwindDataSet.Customers.GetChanges(Data.DataRowState.Added),
            NorthwindDataSet.CustomersDataTable)

        If Not IsNothing(newCustomers) Then
            Try
                CustomersTableAdapter.Update(newCustomers)

            Catch ex As Exception
                MessageBox.Show("AddNewCustomers Failed" & vbCrLf & ex.Message)
            End Try
        End If
    End Sub